File: MoroTest/src/main/java/com/moro/MoroTest/model/UserDetailModel.java
package com.moro.MoroTest.model;

import com.moro.MoroTest.dao.MyUser;
import lombok.Data;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

@Data
public class UserDetailModel implements UserDetails {

    private Long id;
    private String username;
    private String password;
    private List<GrantedAuthority> authorities;

    public UserDetailModel(MyUser user){
        this.id = user.getId();
        this.username = user.getUsername();
        this.password = user.getPassword();

        this.authorities = new ArrayList<>();
        authorities.add(new SimpleGrantedAuthority(user.getRole().name()));
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() { return this.authorities; }

}

File: MoroTestKotlin/src/main/kotlin/com/moro/moroTestKotlin/model/UserDetailModel.kt
package com.moro.moroTestKotlin.model

import com.moro.moroTestKotlin.dao.MyUser
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails


data class UserDetailModel(val user: MyUser) : UserDetails {
    val id: Long = user.id
    private val username: String = user.username
    private val password: String? = user.password

    private val authorities: MutableList<GrantedAuthority> = ArrayList()

    init {
        authorities.add(SimpleGrantedAuthority(user.role.name))
    }

    override fun getAuthorities(): kotlin.collections.Collection<GrantedAuthority> = this.authorities
    override fun getPassword(): String? = this.password
    override fun getUsername(): String = this.username
}